### PR TITLE
Return lease IDs from IDistributedLock

### PR DESCRIPTION
**Pull request by preardon**
Feature #3111

Update Distributed Lock to return and accept a LockId

### CODE REVIEW

**Review by dhickie — APPROVED** [review]

File: Paramore.Brighter.Locking.Azure/AzureBlobLockingProvider.cs [hunk 1]
```diff
@@ -11,9 +11,7 @@ public class AzureBlobLockingProvider(AzureBlobLockingProviderOptions options) :
     private readonly BlobContainerClient _containerClient = new BlobContainerClient(options.BlobContainerUri, options.TokenCredential);
     private readonly ILogger _logger = ApplicationLogging.CreateLogger<AzureBlobLockingProviderOptions>();
 
-    private readonly Dictionary<string, string> _leases = new Dictionary<string, string>();
-
-    public async Task<bool> ObtainLockAsync(string resource, CancellationToken cancellationToken)
+    public async Task<string?> ObtainLockAsync(string resource, CancellationToken cancellationToken)
```
Comment (iancooper) [inline Paramore.Brighter.Locking.Azure/AzureBlobLockingProvider.cs:14]: Our style is to document public methods, using XML comments, as this shows up in Intellisense

File: Paramore.Brighter.Locking.Azure/AzureBlobLockingProvider.cs [hunk 2]
```diff
@@ -11,9 +11,7 @@ public class AzureBlobLockingProvider(AzureBlobLockingProviderOptions options) :
     private readonly BlobContainerClient _containerClient = new BlobContainerClient(options.BlobContainerUri, options.TokenCredential);
```
Comment (iancooper) [inline Paramore.Brighter.Locking.Azure/AzureBlobLockingProvider.cs:11]: Our style is to include the MIT license statement, in a region, in every source code file, but not in tests

**Review by iancooper — CHANGES_REQUESTED** [review]

File: Paramore.Brighter.Locking.Azure/AzureBlobLockingProvider.cs (hunk 1 above)
Comment (preardon) [reply Paramore.Brighter.Locking.Azure/AzureBlobLockingProvider.cs:14]: Thank you, removed

File: Paramore.Brighter.Locking.Azure/AzureBlobLockingProvider.cs [hunk 3]
```diff
@@ -59,41 +56,37 @@ public bool ObtainLock(string resource)
...
-    private BlobLeaseClient? GetBlobLeaseClientForResource(string resource)
+    private BlobLeaseClient? GetBlobLeaseClientForResource(string resource, string lockId)
     {
-        if (_leases.ContainsKey(NormaliseResourceName(resource)))
-            return GetBlobClient(resource).GetBlobLeaseClient(_leases[NormaliseResourceName(resource)]);
+        return GetBlobClient(resource).GetBlobLeaseClient(lockId);
         
-        _logger.LogInformation("No lock found for {LockResourceName}", resource);
-        return null;
+        // _logger.LogInformation("No lock found for {LockResourceName}", resource);
```
[the remark this answers is not in the document]
Comment (preardon) [reply Paramore.Brighter.Locking.Azure/AzureBlobLockingProvider.cs:88]: Thank you, removed

File: Paramore.Brighter.Locking.Azure/AzureBlobLockingProvider.cs (hunk 1 above)
Comment (preardon) [reply Paramore.Brighter.Locking.Azure/AzureBlobLockingProvider.cs:14]: Comments included

File: Paramore.Brighter.Locking.Azure/AzureBlobLockingProvider.cs (hunk 2 above)
Comment (preardon) [reply Paramore.Brighter.Locking.Azure/AzureBlobLockingProvider.cs:11]: Added

**Pull request merged**